dags/openshift_nightlies/scripts/install/rosa.sh

```diff
@@ -137,6 +137,79 @@ _adm_logic_check(){
+    oc get sts prometheus-k8s -n openshift-monitoring
+    echo "Restart stateful set pods"
+    oc rollout restart -n openshift-monitoring statefulset/prometheus-k8s 
+    echo "Wait till they are completely restarted"
+    oc rollout status -n openshift-monitoring statefulset/prometheus-k8s
+    echo "Check pods status again and the hosting nodes"
+    oc get pods -n openshift-monitoring -o wide | grep prometheus-k8s
+    for node in $(oc get pods -n openshift-monitoring -o wide | grep -i prometheus-k8s | grep -i running | awk '{print$7}');
+    do
+        if [[ $(oc get nodes | grep infra | awk '{print$1}' | grep $node) != "" ]]; then
+                echo "$node is an infra node"
+        else
+                echo "ERROR: Prometheus-k8s pod on $node is not an infra node"
+                exit 1
+        fi
+    done
+
+    echo "Initiate migration of ingress router-default pods to infra nodepools"
+    echo "Add toleration to use infra nodes"
+    oc patch ingresscontroller -n openshift-ingress-operator default --type merge --patch  '{"spec":{"nodePlacement":{"nodeSelector":{"matchLabels":{"node-role.kubernetes.io/infra":"true"}},"tolerations":[{"effect":"NoSchedule","key":"node-role.kubernetes.io/infra","operator":"Exists"}]}}}'
+    echo "Wait till it gets rolled out"
+    sleep 60
+    oc get pods -n openshift-ingress -o wide
+    for node in $(oc get pods -n openshift-ingress -o wide | grep -i router | grep -i running | awk '{print$7}');
+    do
+        if [[ $(oc get nodes | grep infra | awk '{print$1}' | grep $node) != "" ]]; then
+                echo "$node is an infra node"
+        else
+                echo "ERROR: router-default pod on $node is not an infra node"
+                exit 1
+        fi
+    done
+    
+    return 0
+}
+
+_wait_for_extra_nodes_ready(){
+    export NODE_LABLES=$(cat ${json_file} | jq -r .extra_machinepool[].labels)
+    for label in $NODE_LABLES;
+    do
+        REPLICA=$(cat ${json_file} | jq -r .extra_machinepool[] | jq -r 'select(.labels == '\"$label\"')'.replica)
+        NODES_COUNT=$((REPLICA*3))
+        _wait_for_nodes_ready $CLUSTER_NAME $NODES_COUNT $label
+        if [[ $label == *"infra"* ]] ; then
+            _balance_infra
+        fi       
+    done
+    return 0
+}
+
+_add_machinepool(){
+    export MACHINEPOOLS=$(cat ${json_file} | jq -r .extra_machinepool[].name)
+    for mcp in $MACHINEPOOLS; 
+    do
+        echo "Add an extra machinepool - $mcp to cluster"
+        REPLICA=$(cat ${json_file} | jq -r .extra_machinepool[] | jq -r 'select(.name == '\"$mcp\"')'.replica)
+        INS_TYPE=$(cat ${json_file} | jq -r .extra_machinepool[] | jq -r 'select(.name == '\"$mcp\"')'.instance_type)
+        LABELS=$(cat ${json_file} | jq -r .extra_machinepool[] | jq -r 'select(.name == '\"$mcp\"')'.labels)
+        TAINTS=$(cat ${json_file} | jq -r .extra_machinepool[] | jq -r 'select(.name == '\"$mcp\"')'.taints)
+        for ZONE in a b c;
```

Comment:
We actually need only 2 infra nodes. There's no need to create 3 machine pools

Reply:
Did ROSA change to 2 as well?

Reply:
Classic ROSA still creates 3 Infra, but on hosted-cp we can control them using machinepool size.
@rsevilla87 is 2 infra enough for hosting 2 x prom-k8s as well  as 2 x router pods ?